### PR TITLE
Add object/flatten, closes #221.

### LINKED ISSUE #221: object/flatten
I think this would be a good addition, similar to array/flatten but for objects.

If everyone agrees, I can do a PR with it. I already have a working function in a project.

### PR DESCRIPTION
**Pull request by satazor**
Made also a minor optimization in the array/flatten.

### CODE REVIEW

**Comment by satazor** [comment]
@millermedeiros tests are failing in date/diff tests (have node 0.12).

**Comment by satazor** [comment]
Oops, my mistake. Will make a new PR.

File: tests/spec/object/spec-flatten.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            result = flatten(obj, 1);
+            expect(result).toEqual({ a: 1, b: 2, 'c.d': 3, 'c.e': { f: 4 } });
+        });
+
+        it('should only flatten 2 layers if level is 2', function () {
+            var obj = { a: 1, b: 2, c: { d: 3, e: { f: 4, g: { h: 5 } } } },
+                result;
+
+            result = flatten(obj, 2);
+            expect(result).toEqual({ a: 1, b: 2, 'c.d': 3, 'c.e.f': 4, 'c.e.g': { h: 5 } });
```
Comment (jdalton) [inline tests/spec/object/spec-flatten.js]: Seems kind of odd to assume props would be handled as `c[dot]e[dot]f`. Is there another implementation this is based off of?

Comment (satazor) [inline tests/spec/object/spec-flatten.js]: I've merged this by accident, lets discuss this in #224

**Pull request merged**